Modules/Module_Processing/xASL_init_LoadDataPar.m

```diff
@@ -134,4 +136,61 @@
     %% Load dataPar
     x = xASL_adm_MergeStructs(dataPar.x, x);
     
-end
\ No newline at end of file
+end
+
+%% -----------------------------------------------------------------------------
+%% -----------------------------------------------------------------------------
+function [x] = xASL_initLoadDataPar_PrepareAtlas4ROI(x)
+%Check for x.S.Atlases & x.S.TissueMasking and provide default values or print instructions
+
+bAtlasTissueMatch = true; % Necessary condition
+
+if ~isfield(x, 'S') || (~isfield(x.S,'Atlases') && ~isfield(x.S, 'TissueMasking'))
+	% Default atlases/ROIs & tissue masks if nothing is provided
+	x.S.Atlases = {'TotalGM','DeepWM'}; % Default
+    x.S.TissueMasking = {'GM' 'WM'}; % GM WM, fits with the TotalGM & DeepWM above
+    % Note that this should be in the same order as the atlases/ROIs
+    % A mismatch (e.g. TissueMasking=GM for Atlases=deepWM) would result in an empty ROI, producing a NaN in the .tsv table
+elseif ~isfield(x.S, 'Atlases') && isfield(x.S, 'TissueMasking')
+	% Missing Atlases, but provided TissueMasking - cannot continue
+    warning('Custom tissue-types (x.S.TissueMasking) specified without ROI atlas-selection (x.S.Atlases). Atlases need to be provided. See instructions below:');
+    bAtlasTissueMatch = false;
+elseif isfield(x.S, 'Atlases') && isfield(x.S, 'TissueMasking') && length(x.S.Atlases)~=length(x.S.TissueMasking)
+	% Non matching lengths, cannot continue
+    warning('The same number of ROI atlases as subject-wise tissue-types provided does not match:');
+    fprintf('%s\n', ['S:{Atlases:["' strjoin(x.S.Atlases, '", "') '"]}']);
+    fprintf('%s\n', ['S:{TissueMasking:["' strjoin(x.S.TissueMasking, '", "') '"]}']);
+    bAtlasTissueMatch = false;
+elseif 	isfield(x.S, 'Atlases') && ~isfield(x.S, 'TissueMasking')
+	% TissueMasking not provided, so it has to be extracted from Atlases as previously.
+    warning('ROIs provided in S.Atlases without the tissue-types for these ROIs in S.TissueMasking.');
+    bAtlasTissueMatch = false;
+	textRecommendation = 'S:{TissueMasking:["';
+	% Fill in TissueMasking based on the atlas names and default to GM. The user will see a warning and automatic tissue masks to verify
+	for iAtlas = 1:numel(x.S.Atlases)
+		if iAtlas > 1
+			textRecommendation = [textRecommendation '", "'];
+		end
+		if ~isempty(regexpi(x.S.Atlases{iAtlas}, 'WM')) || ~isempty(regexpi(x.S.Atlases{iAtlas}, 'whitematter'))
```

Comment:
Nice, though we should be careful as the ignorant user may just copy what we write here which may not be correct. 

Note that the only case where this can be inferred from our ROI atlases is "DeepWM" (now totalGM==a WholeBrain mask already and we should rename in this issue).

For other provided atlases it is risky to infer this. E.g., most people assume that we provide GM values, so if they provide a custom atlas named "HarvardOxfordWholeBrain", they would assume that we analyze GM. They would fill in "WB" as per your recommendation and still think ExploreASL provided them GM CBF.

So, I wonder if we shouldn't just remove this here, and link to the tutorial (which can be a specific part of the tutorial where we explain this). And then I would explain "if you want to analyze GM, write "GM", etc, not 

"as example for an atlas ending with WB, you should write WB"

Reply:
Yes, we should write this carefully and explain them the caveats. But to make their life easier, I also want to prepare the exact line they should add. So I'd like to keep the example that is adjusted to the correct length. Otherwise, they keep struggling with putting wrong lengths etc. Please suggest a text that explains that they should check the tissue types. Note that this recommendation is printed addiotional to referring them to the manual and additional to printing the instructions. So without this suggestion, the ignorant user will keep copy-pasting the example from the manual...

Reply:
warning('x.S.TissueMasking input did not match x.S.Atlases');
fprintf(%s\n, 'If you want GM CBF from an atlas ROI, add '''GM'''. If you want WM CBF from an atlas ROI, add '''WM'''');
fprintf(%s\n, 'If you want both GM CBF and WM CBF from an atlas ROI, add the atlas ROI twice and add both '''GM''' and '''WM'''');
fprintf(%s\n, 'E.g., if you want both GM CBF and WM CBF from the Harvard-Oxford atlas, you would provide the following:');
fprintf(%s\n, 'x.S.Atlases = {''''HOConn''' ''''HOConn'''});
fprintf(%s\n, 'x.S.TissueMasking = {''''GM''' ''''WM'''});

Or something similar.
My main point is that we cannot advice the user based on trying to detect what they want based on the atlas name, we should remove the regexpi(x.S.Atlases{iAtlas}) etc.

We rather want the user to puzzle more or ask us, then somebody reporting WM CBF as GM CBF because they just copy-pasted and didn't understand it.

Reply:
Warning added as proposed.